src/Services/MailFinderService.php

```diff
@@ -82,24 +76,28 @@ public function findTemplateByTechnicalName(
 
         $searchFolder = $this->searchPathProvider->buildPaths($businessEvent);
 
-        $criteria = (new Criteria)->addFilter(new EqualsFilter('salesChannels.id', $businessEvent->getSalesChannelId()));
-        $searchResult = $this->themeRepository->search($criteria, $businessEvent->getContext());
+        $themePath = $this->connection->fetchOne(
+            'SELECT IFNULL(a.path, p.path) AS `path` FROM `theme` AS t '
+                    . 'LEFT JOIN `theme_sales_channel` AS tsc ON tsc.`theme_id` = t.`id` '
+                    . 'LEFT JOIN `plugin` AS p ON p.`name` = t.`technical_name` '
+                    . 'LEFT JOIN `app` AS a ON a.`name` = t.`technical_name` '
+                    . 'WHERE tsc.`sales_channel_id` = UNHEX("' . $businessEvent->getSalesChannelId() . '");'
```

Comment:
Can you use prepared statements? Uuid::fromHexToBytes helps you also to convert it